### PR TITLE
f-checkout@0.27.0 - vuex mutations and actions for input fields updates

### PR DESCRIPTION
**Pull request by dandel10n**
### Changed
- Replaced direct state manipulations with mutations and actions

It is still wip, but I would appreciate some feedback.
Test should be added to this PR shortly

I am fine with upcoming repo restructuring in the end of December, I will move my changes to the fresh master in January

### CODE REVIEW

File: packages/f-checkout/src/components/Address.vue [hunk 1]
```diff
@@ -115,13 +119,36 @@ export default {
...
         /*
         * Returns true if `field` has been touched and if it is still empty
         * The $dirty boolean changes to true when the user has focused/lost
         * focus on the input field.
         */
         isFieldEmpty (field) {
             return this.$v.addressValidations[field].$dirty && !this.$v.addressValidations[field].required;
+        },
+        updateAddressLine1(value) {
+            this.$store.dispatch('checkout/updateFulfilmentAddress', {
```
Comment (DamianMullins) [inline packages/f-checkout/src/components/Address.vue]: You should be able to use the mapped action here and below?

```suggestion
            this.updateFulfilmentAddress({
```

File: packages/f-checkout/src/components/Checkout.vue [hunk 2]
```diff
@@ -345,6 +346,9 @@ export default {
         */
         isValidPostcode () {
             return validations.isValidPostcode(this.fulfilment.address.postcode, this.$i18n.locale);
+        },
+        updateMobileNumber (value) {
+            this.$store.dispatch('checkout/updateMobileNumber', value)
```
Comment (DamianMullins) [inline packages/f-checkout/src/components/Checkout.vue]: Should be able to map the `updateMobileNumber` action in `mapActions` above and call the action directly?

File: packages/f-checkout/src/store/checkout.module.js [hunk 3]
```diff
@@ -85,12 +85,18 @@ export default {
         },
 
         setAuthToken: ({ commit }, authToken) => {
-            commit('updateAuth', authToken);
+            commit('UPDATE_AUTH', authToken);
+        },
+        updateFulfilmentAddress ({ commit }, payload) {
```
Comment (DamianMullins) [inline packages/f-checkout/src/store/checkout.module.js:90]: NIT: Add a space between actions.

File: packages/f-checkout/src/store/checkout.module.js [hunk 4]
```diff
@@ -126,9 +132,18 @@ export default {
             state.messages = messages;
         },
 
-        updateAuth: (state, authToken) => {
+        UPDATE_AUTH: (state, authToken) => {
             state.authToken = authToken;
             state.isLoggedIn = !!authToken;
+        },
```
Comment (DamianMullins) [inline packages/f-checkout/src/store/checkout.module.js:138]: NIT: Add a space between mutations.

File: packages/f-checkout/src/components/Address.vue [hunk 5]
```diff
@@ -115,13 +119,36 @@ export default {
...
+        ]),
         /*
         * Returns true if `field` has been touched and if it is still empty
         * The $dirty boolean changes to true when the user has focused/lost
         * focus on the input field.
         */
         isFieldEmpty (field) {
             return this.$v.addressValidations[field].$dirty && !this.$v.addressValidations[field].required;
+        },
+        updateAddressLine1(value) {
```
Comment (matthewhardern) [inline packages/f-checkout/src/components/Address.vue:133]: Question should the methods have jsdoc comments?

**Review by ashleynolan — CHANGES_REQUESTED** [review]
Marking this as do not merge. This PR will need refactoring once the folder structure is changed (probably easier to make a copy of the changes you have made and re-apply them to the new locations, as merging will likely give you a bunch of conflicts)

**Pull request closed without merging**